staging/src/k8s.io/apiserver/pkg/endpoints/handlers/rest.go

```diff
@@ -390,6 +391,12 @@ func limitedReadBody(req *http.Request, limit int64) ([]byte, error) {
 	return data, nil
 }
 
+func limitedReadBodyWithRecordMetric(ctx context.Context, req *http.Request, limit int64, resourceGroup string, verb requestmetrics.RequestBodyVerb) ([]byte, error) {
+	readBody, err := limitedReadBody(req, limit)
+	requestmetrics.RecordRequestBodySize(ctx, resourceGroup, verb, len(readBody))
```

Comment:
Should you log `if err != nil` since in that case I would expect `len(readBody)` to be 0, which will mess-up your metric.